xcdat/spatial.py

```diff
@@ -301,14 +305,10 @@ def _validate_axis_arg(self, axis: List[SpatialAxis]):
                     f"{', '.join(SPATIAL_AXES)}."
                 )
 
-            try:
-                self._dataset.cf.axes[key]
-            except KeyError:
-                raise KeyError(
-                    f"A '{key}' axis dimension was not found in the dataset. Make sure "
-                    f"the dataset has '{key}' axis coordinates and the coordinates' "
-                    f"'axis' attribute is set to '{key}'."
-                )
+            coord_var = get_coord_var(self._dataset, key)
+            # Set the attribute for the axis to the coordinate variable if the
+            # coordinate variable exists.
+            self.__setattr__(f"_{key}", coord_var)
```

Comment:
@tomvothecoder - I didn't understand why you were setting an attribute (e.g., `_lat`). I noticed this because when I tried to call `xcdat.spatial._averager()` directly it gave me an error 

> AttributeError: 'Dataset' object has no attribute '_lat'

Because I had not yet ran `_validate_axis_arg()` (which sets these attributes). I'm sure there is a good reason for this...is there a **simple** explanation for setting these underscore attributes?

Reply:
This makes sense. One consequence is that I cannot call `_averager()` directly even if I have the inputs outlined in the documentation. I think that is probably okay because this is not intended to be a public method.

Reply:
Continuing to work on this. I'm now wondering if similar changes need to be made to the `_validate_axis_arg` in `bounds.py`? I'm now getting a KeyError in `axis.py` at this line.

```
    bounds = xr.concat(
        (bounds, bounds[p_meridian_index, :]), dim=bounds.cf.axes["X"][0]
    )
```

I get an empty dictionary for this dim argument: 

```
print(bounds.cf.axes)
```

> {}

Reply:
I ended up removing the implementation of the underscore attributes because we only need the dimension name(s) for the `axis` arg a few times. It also makes the code much easier to read compared to  using `__setattr__` and `getattr` for the underscore attributes.

I implemented `get_axis_dim()` for this specific metadata:
https://github.com/xCDAT/xcdat/blob/93a4a7fff990bbac9d0bafb57a472c7950e95ae5/xcdat/axis.py#L86-L99

Now in `_average()`:
https://github.com/xCDAT/xcdat/blob/93a4a7fff990bbac9d0bafb57a472c7950e95ae5/xcdat/spatial.py#L723-L732

You should be able to test `_average()` without needing to set the underscore attributes.

I also fixed `bounds._validate_axis_arg()`:
https://github.com/xCDAT/xcdat/blob/93a4a7fff990bbac9d0bafb57a472c7950e95ae5/xcdat/axis.py#L344-L348

Reply:
Marking this as resolved.